docs/sequence/PISPTransactionApi.puml

```diff
@@ -71,78 +76,214 @@ rnote right of PISP #Light
+  transactionRequestState: 'RECEIVED'
+}
+end note
+3p_DFSP_IBS -> Switch: PUT /thirdpartyRequests/transaction/{id}
+activate Switch
+Switch --> 3p_DFSP_IBS: 202 Accepted
+Switch -> 3p_PISP_IBS: PUT /thirdpartyRequests/transaction/{id}
+activate 3p_PISP_IBS
+3p_PISP_IBS --> Switch: 202 accepted
+deactivate Switch
+3p_PISP_IBS -> ST8: transactionRequestReceived()
+activate ST8
+rnote right of ST8: state: transactionInitiated
+ST8 -> ST8: store received **transactionId**
+rnote right of ST8: state: pendingTransactionAuthorization
+deactivate ST8
+deactivate 3p_PISP_IBS
 
-Switch -> IBS: ""POST /authorizations""
-IBS -> ST8: ""onPostAuthorizations()""
-IBS --> Switch: ""200 OK""
-rnote right of ST8: ""state: authorizationReceived""
+note over 3p_DFSP_IBS
+  DFSP requires QuotesInformation\n
+end note
+rnote right of 3p_DFSP_IBS: state: pendingQuotesRequest
+3p_DFSP_IBS -> SDK_DFSP_OBS: SYNC POST /quotes
+activate SDK_DFSP_OBS
+SDK_DFSP_OBS -> SDK_DFSP_OBS: retrieve QuotesInformation from Switch
+SDK_DFSP_OBS -> 3p_DFSP_IBS: 200 QuotesInformation
+deactivate SDK_DFSP_OBS
+note over 3p_DFSP_IBS
+  DFSP has the QuotesInformation, they can now ask the PISP for authorization
+  **transactionRequestState** is **PENDING**
+end note
+rnote right of 3p_DFSP_IBS: state: pendingAuthorizationRequest
+3p_DFSP_IBS -> SDK_DFSP_OBS: SYNC POST /authorization
+activate SDK_DFSP_OBS
+SDK_DFSP_OBS -> Switch: POST /authorization
+activate Switch
+Switch --> SDK_DFSP_OBS: 202 Accepted
+SDK_DFSP_OBS -> SDK_DFSP_OBS: waiting on PUT /authorization/{id}
+deactivate SDK_DFSP_OBS
+Switch -> 3p_PISP_IBS: POST /authorizations
+activate 3p_PISP_IBS
+3p_PISP_IBS --> Switch: 200 OK
+deactivate Switch
+3p_PISP_IBS -> ST8: authorizationRequestReceived()
+activate ST8
+deactivate 3p_PISP_IBS
+rnote right of ST8: state: authorizationReceived
 
 
-ST8 -> OBS: ""something...""
+ST8 -> 3p_PISP_OBS: QuotesInformation, transactionId
 deactivate ST8
+deactivate 3p_PISP_IBS
 
 'TODO: add authorization details
-OBS --> PISP: ""200 OK""
-deactivate OBS
+3p_PISP_OBS --> PISP: 200 OK
+deactivate 3p_PISP_OBS
 deactivate PISP
 
 ...Present quote to be approved\nUser signs the challenge...
 
 
 == Approve Transaction ==
 rnote right of PISP #Light
-""POST /thirdpartyTransactions/{id}/approve""
+POST /thirdpartyTransactions/{id}/approve
 {
   signedChallenge: "base64(xxx)"
 }
 end note
-PISP -> OBS: ""POST /thirdpartyTransactions/{id}/approve""
+PISP -> 3p_PISP_OBS: POST /thirdpartyTransactions/{id}/approve
 activate PISP
-activate OBS
+activate 3p_PISP_OBS
 
-OBS -> ST8: ""approve()""
+3p_PISP_OBS -> ST8: approve()
 activate ST8
-rnote right of ST8: ""state: approvalReceived""
+rnote right of ST8: state: approvalReceived
 
 
-ST8 -> Switch: ""PUT /authorizations/{id}""
-Switch --> ST8: ""202 Accepted""
-rnote right of ST8: ""state: pendingTransaction""
+ST8 -> Switch: PUT /authorizations/{id}
+activate Switch
+Switch --> ST8: 202 Accepted
+rnote right of ST8: state: pendingTransaction
```

Comment:
could this be something like `pendingTransactionFinalization` or something? `pendingTransaction` seems a little vague to me.

Reply:
I renamed it to `pendingVerifyAuthorization`